test/service-worker_test.js

```diff
@@ -64,21 +64,30 @@ suite('service-worker', () => {
   suite('generateServiceWorker()', () => {
 
     test('should throw when options are not provided', () => {
-      assert.throws(() => {
-        serviceWorker.generateServiceWorker();
-      }, Error.AssertionError, '`project` & `buildRoot` options are required');
+      return serviceWorker.generateServiceWorker()
+        .then(() => { throw new Error('generateServiceWorker() resolved, expected rejection!'); })
```

Comment:
I would do Promise inversions as two callback to `then()`, so that you don't have to worry about the exception you throw in `then()` flowing into `catch()`. Then just assert.fail() in the first callback. Alternatively, we have a handy `invert()` function hangout out around here somewhere...